main/ecrecover/FPSECP256K1/addFpSecp256k1.zkasm

```diff
@@ -9,21 +9,13 @@
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
 ; RESOURCES:
-;   2 ariths + 7 steps
+;   1 arith + 4 steps
 
 addFpSecp256k1:
 
-        ; 1] Compute and check the sum over Z
-        ; A·[1] + C = [D]·2²⁵⁶ + [E]
+        ; A·1 + C = op (mod SECP256K1_P)
         1 => B
-        $${var _addFpSecp256k1_AC = A + C}
-        ${_addFpSecp256k1_AC >> 256} => D
-        ${_addFpSecp256k1_AC} => E :ARITH
+        %SECP256K1_P => D
+        ${(A*B + C) % D} => E   :ARITH_MOD
```

Comment:
optimization (-1 step):
; C changes in row after ARITH and executor no need to calculate A*B,
${(A + C) % D} => C :ARITH_MOD, RETURN

Reply:
Added